File: Programmers/Level3/noOvertime(jpct)/noOvertime.py
def noOvertime(n, works):
    for i in range(n):
    	if len(works) == 0:
    		break;
    	works.sort(reverse=True)
    	works[0]-=1
    	while works.count(0)!=0:
	    	works.remove(0)
    return sum([num**2 for num in works])

n = 10
works = [2, 0, 3, 9,7,7,2,5]
print(noOvertime(n,works))

File: Programmers/Level3/noOvertime(jpct)/NoOverTime.java
class NoOvertime {
	public int noOvertime(int no, int[] works) {
		int result = 0;
		int sum = 0;
		for(int i=0; i<works.length; ++i)
		{
			sum += works[i];
		}

		if(sum <= no)
		{
			return 0;
		}
		int maximum = works[0];
		while(no>0)
		{
			for(int i=0;i<works.length;++i)
			{
				if(maximum < works[i])
				{
					maximum = works[i];
				}
			}

			for(int i=0; i<works.length;++i)
			{
				if(maximum == works[i])
				{
					--maximum;
					--works[i];
					--no;
					break;
				}
			}
		}
		
		for(int i=0; i<works.length; ++i)
		{
			result += works[i] * works[i];
		}
		return result;
	}

	public static void main(String[] args) {
		NoOvertime c = new NoOvertime();
		int []test = {4,3,3};
		System.out.println(c.noOvertime(4,test));
	}
}